plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/NewApplicationWizardModel.java

```diff
@@ -207,29 +222,58 @@ public void loadResources() {
 
 	@Override
 	public void setUseLocalAppSource(boolean useLocalTemplate) {
-		update(useLocalTemplate, this.project, this.projectItems, this.serverAppSource, this.localAppSourceFilename);
+		update(useLocalTemplate, this.project, this.projectItems, this.serverAppSource, this.localAppSourceFilename, this.appSourceStatus);
 	}
 
 	@Override
 	public boolean isUseLocalAppSource() {
 		return useLocalAppSource;
 	}
 
-	public InputStream createInputStream(String fileName) throws FileNotFoundException {
-		return new FileInputStream(fileName);
+	public InputStream createInputStream(String filename, IProgressMonitor monitor) throws IOException {
+	    String url = OpenshiftUIConstants.URL_VALIDATOR.isValid(filename)?filename:new File(filename).toURI().toString();
+	    try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
```

Comment:
I'd rather prefer we only call OpenshiftUIConstants.TRANSPORT_UTILITY.download if it's a valid URL, else expect a file and return a fileinputstream directly